block/sync.go

```diff
@@ -0,0 +1,134 @@
+		return
+	}
+
+	// Update heights based on state update end height
+	m.LastSettlementHeight.Store(eventData.EndHeight)
+
+	m.logger.Error("syncing")
+
+	// Update sequencers list from SL
+	err := m.UpdateSequencerSetFromSL()
+	if err != nil {
+		m.logger.Error("update bonded sequencer set", "error", err)
+	}
+
+	if eventData.EndHeight > m.State.Height() {
+		// trigger syncing from settlement last state update.
+		m.triggerSettlementSyncing()
+		// update target height used for syncing status rpc
+		m.UpdateTargetHeight(eventData.EndHeight)
+
+	} else {
+		// trigger validation of the last state update available in settlement
+		m.triggerSettlementValidation()
+	}
+}
+
+// SettlementSyncLoop listens for syncing triggers which indicate new settlement height updates, and attempts to sync to the last seen settlement height.
+// Syncing triggers can be called when a new settlement state update event arrives or explicitly from the `updateFromLastSettlementState` method which is only being called upon startup.
+// Upon new trigger, we know the settlement reached a new height we haven't seen before so a validation signal is sent to validate the settlement batch.
+
+// Note: even when a sync is triggered, there is no guarantee that the batch will be applied from settlement as there is a race condition with the p2p/blocksync for syncing.
+func (m *Manager) SettlementSyncLoop(ctx context.Context) error {
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+
+		case <-m.settlementSyncingC:
+
+			m.logger.Info("syncing to target height", "targetHeight", m.LastSettlementHeight.Load())
+
+			for currH := m.State.NextHeight(); currH <= m.LastSettlementHeight.Load(); currH = m.State.NextHeight() {
+				// if we have the block locally, we don't need to fetch it from the DA.
+				// it will only happen in case of rollback.
+				err := m.applyLocalBlock(currH)
+				if err == nil {
+					m.logger.Info("Synced from local", "store height", currH, "target height", m.LastSettlementHeight.Load())
+					continue
+				}
+				if !errors.Is(err, gerrc.ErrNotFound) {
+					m.logger.Error("Apply local block", "err", err)
+				}
+
+				settlementBatch, err := m.SLClient.GetBatchAtHeight(m.State.NextHeight())
+				if err != nil {
+					return fmt.Errorf("retrieve batch: %w", err)
+				}
+				m.logger.Info("Retrieved state update from SL.", "state_index", settlementBatch.StateIndex)
+
+				err = m.ApplyFromSLBatch(settlementBatch.MetaData.DA)
+				if err != nil {
+					m.logger.Error("process next DA batch", "err", err)
+				}
+
+				// if height havent been updated, we are stuck
+				if m.State.NextHeight() == currH {
+					return fmt.Errorf("stuck at height %d", currH)
+				}
+
+				m.logger.Info("Synced from DA", "store height", m.State.Height(), "target height", m.LastSettlementHeight.Load())
+
+				// trigger state update validation, after each state update is applied
+				m.triggerSettlementValidation()
```

Comment:
I see we trigger this from both when state update is available on `onNewStateUpdate` and also here. 
do we really need to trigger it from both places? isn't it covering both?

Reply:
we called it here because in case there is a new state update and the node is not synced, first it will sync (without validating), and then validate (from this call). 
if the node is already synced on a new state update, the syncing channel will not be called and it will only validate.